.github/workflows/deployment.yml

```diff
@@ -212,7 +212,7 @@ jobs:
       - name: Checkout cli/cli
         uses: actions/checkout@v4
       - name: Merge built artifacts
-        uses: actions/download-artifact@v3
```

Comment:
@williammartin : Reviewing [notes behind the major version change](https://github.com/actions/toolkit/tree/main/packages/artifact), I think we would be okay upgrading, however I think there is an additional change to `actions/upload-artifact` to use `@v4` too

**What are your thoughts based on the details below?**

> The release of @actions/artifact@v2 (including upload-artifact@v4 and download-artifact@v4) are major changes to the backend architecture of Artifacts. They have numerous performance and behavioral improvements.
> 
> Improvements
> 1. All upload and download operations are much quicker, up to 80% faster download times and 96% faster upload times in worst case scenarios.
> 1. Once uploaded, an Artifact ID is returned and Artifacts are immediately available in the UI and [REST API](https://docs.github.com/en/rest/actions/artifacts). Previously, you would have to wait for the run to be completed before an ID was available or any APIs could be utilized.
> 1. Artifacts can now be downloaded and deleted from the UI before the entire workflow run finishes.
> 1. The contents of an Artifact are uploaded together into an immutable archive. They cannot be altered by subsequent jobs. Both of these factors help reduce the possibility of accidentally corrupting Artifact files. (Digest/integrity hash coming soon in the API!)
> 1. This library (and actions/download-artifact) now support downloading Artifacts from other repositories and runs if a GITHUB_TOKEN with sufficient actions:read permissions are provided.
>
> Breaking changes
> 1. Firewall rules required for self-hosted runners.
> 
>    If you are using self-hosted runners behind a firewall, you must have flows open to [Actions endpoints](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#communication-between-self-hosted-runners-and-github). If you cannot use wildcard rules for your firewall, see the GitHub [meta endpoint](https://api.github.com/meta) for specific endpoints.
> 
>    e.g.
> 
>    ```
>    curl https://api.github.com/meta | jq .domains.actions
>    ```
> 1. Uploading to the same named Artifact multiple times.
> 
>    Due to how Artifacts are created in this new version, it is no longer possible to upload to the same named Artifact multiple times. You must either split the uploads into multiple Artifacts with different names, or only upload once.
> 
> 1. Limit of Artifacts for an individual job.
> 
>    Each job in a workflow run now has a limit of 10 artifacts.

Reply:
How would we know if it hasn't worked as expected? Generally seems fine to me though.